system/churchdb/upload.php

```diff
@@ -144,10 +144,14 @@ function handleUpload($uploadDirectory, $replaceOldFile = FALSE){
 
         if ($this->file->save($uploadDirectory . $filename . '.' . $ext)){
 
-
-          $person_id = $_GET['userid'];
-          $filename_old=$uploadDirectory . $filename . '.' . $ext;
-          $filename=md5(uniqid()).'.'.$ext;
+          $filename_old = $uploadDirectory . $filename . '.' . $ext;
+            
+          if (function_exists('openssl_random_pseudo_bytes')) {
+              $filename = bin2hex(openssl_random_pseudo_bytes(32)).'.'.$ext;
+          } else {
+              $rands = array_map('mt_rand', array_fill(0, 10, 0), array_fill(0, 10, mt_getrandmax()));
+              $filename = md5(implode($rands)).'.'.$ext;
```

Comment:
BTW: The "real" randomness of the mt_rand() fallback is only 32 bits anyway, because the Mersenne Twister is only seeded with 4 bytes on PHP init, so it's really no more than enough to be sufficiently random...